File: produto.js
class produto {
    constructor(id, descricao, qtde, preco){
        this.id = id
        this.descricao = descricao
        this.qtde = qtde
        this.preco = preco
    }

    vender = (x) => {
        this.qtde = this.qtde + x
    }
    comprar = (x) => {
        if((this.qtde - x) >= 0){
            this.qtde = this.qtde - x
        }
        else{
            console.log(`Preco nao pode ser negativo`)
        }
    }
    aumentar = (x) => {
        this.preco += x
    }
    diminuir = (x) => {
        if(this.preco - x >= 0){
            this.preco = this.preco - x
        }
        else{
            console.log(`Preco nao pode ser negativo`)
        }
    }
    mostra = () => {
        console.log(`${this.descricao} tem ${this.qtde} com preco de R$ ${this.preco}`)
    }
}


let produto1 = new produto(1, "Alcool em gel", 10, 20)
produto1.comprar(10)
produto1.aumentar(2)
produto1.mostra()

let produto2 = new produto(2, "Mascara", 50, 5)
produto2.vender(5)
produto2.diminuir(4)
produto2.mostra()

File: NETBEANS_PROJECTS_JV/Lista1/src/main/java/br/com/unifacef/lista1/ex3/Produto.java

package br.com.unifacef.lista1.ex3;

public class Produto {
    int id;
    int qtde;
    String descricao;
    float preco;
    
    public Produto(){
    }
    
    public Produto(int id, int qtde, String descricao, float preco){
        this.id = id;
        this.qtde = qtde;
        this.descricao = descricao;     
        this.preco = preco;   
    }
    
    public void comprar(int x){
        this.qtde = this.qtde + x;
    }
    
    public void vender(int x){
        if (this.qtde - x >= 0){  
                this.qtde = this.qtde - x;
            }
            else {
                System.out.println("Impossivel, não possui essa quantidade no estoque! \n");
            }
    }
    
    public void subir(float x){
        this.preco = this.preco + x;
    } 
    
    public void descer(float x){
        if (this.preco - x >= 0){  
                this.preco = this.preco - x;
            }
            else {
                System.out.println("Impossivel, preço ficará negativo!");
            }
    }
    
    public String mostra(){
        return "Nome do produto : " + this.descricao + "\n ID : " + this.id + "\n Quantidade : " + this.qtde + "\n Valor : " + this.preco + "\n";
    }
    
}
